File: Misc-Maths/maths.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

int LOOP_CNT = 2;

int i, rand_num, the_ans, ans = 0;
int x;
int y;
time_t start, finish;
FILE *fp;


int main(){
  while (i < LOOP_CNT) {
	  int tmp = 0;
          srand(time(NULL));
          x = rand() % 9 + 1;
          y = rand() % 9 + 1;
          rand_num = rand() % 4 + 1; 
          time(&start); 
    if (rand_num == 1) {
            printf("What is %d + %d ? ", x, y);
            ans = x + y;
    } else if (rand_num == 2){
	    if (x < y) {
		    tmp = x;
		    x = y;
		    y = tmp;
            }
	    printf("What is %d - %d ? ", x, y);
	    ans = x - y;
    } else if (rand_num == 3) {
	    printf("What is %d * %d ? ", x, y);
	    ans = x * y;
    } else {
	    printf("What is %d squared? ", x);
	    ans = x * x;
    }
    fflush(stdout);
    i++;
    scanf("%d", &the_ans); 
    time(&finish);
    if (ans != the_ans){
	    printf("\nC'mon man, we're outta here!\n");
	    exit(1);
    } else if (difftime(finish, start) > 3) {
	    printf("\nAwww man, you ran out of time\n");
	    exit(2);
    }
    fflush(stdout);
  }
fp = fopen("./flag.txt", "r");

if (fp == NULL) {
	printf("\nYou need to put a dummy flag.txt file\n");
	exit(3);
}

char ch;
printf("\nHell yeah!  You're a Math-lete!  Have a flag: ");
while ((ch = fgetc(fp)) != EOF) {
	printf("%c", ch);
}

printf("\n\n");
fclose(fp);

}

File: CTFs/teractf2024/Misc-Maths/maths.py
#!/usr/bin/env python3

import random
from datetime import datetime
import time

random.seed(int(datetime.now().timestamp()))
i = 0
while i <= 2000:
    # print(f"random is {random.randint(1,4)}")
    x = random.randint(0,9)
    y = random.randint(0,9)
    func = random.randint(1,4)
    start = datetime.now().timestamp()
    if func == 1:
        print(f"What is {x} + {y} ? ", end="")
        ans = x + y
    elif func == 2:
        if x < y:
            x, y = y, x
        print(f"What is {x} - {y} ? ", end="")
        ans = x - y
    elif func == 3:
        print(f"What is {x} x {y} ? ", end="")
        ans = x * y
    else:
        print(f"What is {x} squared ? ", end="")
        ans = x ** 2

    i = i + 1 

    if ans != int(input("")):
        print(f"Common man, we're outta here")
        exit(1)
    elif datetime.now().timestamp() - start > 3:
        print(f"Ah man, you ran out of time")
        exit(2)
    
print("Heck yeah!  Have a flag!: ", end="")

#for i in range(50):
#    print(f'this is randint: {random.randint(1,4)}')
